### PR TITLE
Packit config file (.packit.yaml)

### PR DESCRIPTION
**Pull request by jpopelka**
[Packit](https://packit.dev) is a [Github App](https://github.com/marketplace/packit-as-a-service) which helps you integrate this project/package downstream (Fedora).

Just add a config (i.e. merge this) and install the [Github App](https://github.com/marketplace/packit-as-a-service) in this repo.
There are several [jobs](https://packit.dev/docs/configuration/#supported-jobs) in the config file, which tell the service what to do:
- `[copr_build]` anytime a PR is created/updated, Packit builds RPM packages in [copr](https://github.com/marketplace/packit-as-a-service) and lets you know whether everything's still ok.
- `[propose_downstream]` (not implemented yet, but soon :-) ) when a release is created, Packit creates PR's in [Fedora dist-git](https://src.fedoraproject.org/rpms/python-dockerfile-parse)
- `[sync_from_downstream]` picks up a change (mass rebuild, proven packager rebuild or a fix) from Fedora dist-git and sends it to this repo

### CODE REVIEW

**Comment by TomasTomecek** [comment]
Sadly `packit srpm` fails here because of https://github.com/packit-service/packit/issues/480 https://github.com/packit-service/packit/issues/481

**Comment by MartinBasti** [comment]
Thank you packit configuration.

Please let us know when this PR will be ready for review

**Comment by TomasTomecek** [comment]
about to cut packit 0.6.0 - we should be able to fix this PR using that version of packit

**Comment by jpopelka** [comment]
Should be ready for review now.

As I said in the description packit is mainly a Github App, but we also have a CLI, which you can try (before you merge this):
- have `.packit.yaml` in the dockerfile-parse dir (that's what this PR is about)
- `dnf --enablerepo=updates-testing install packit`
- make sure you have [copr token](https://copr.fedorainfracloud.org/api/)
- `packit copr-build --nowait` - last line will tell you url, where you can see whether it builds in copr in all the targets specified in `.packit.yaml`

Merge this PR, install the [Github App](https://github.com/marketplace/packit-as-a-service) in this repo and we can see whether it works in another PR(#84).

You can check also [our guide](https://packit.dev/docs/guide/).

File: .packit.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    - python-dockerfile-parse.spec
+    - .packit.yaml
+
+# name in upstream package repository/registry (e.g. in PyPI)
+upstream_project_name: dockerfile-parse
+# downstream (Fedora) RPM package name
+downstream_package_name: python-dockerfile-parse
+
+current_version_command: ["python3", "setup.py", "--version"]
+create_tarball_command: ["python3", "setup.py", "sdist", "--dist-dir", "."]
```
Comment (TomasTomecek) [inline .packit.yaml]: it works for me even with default version and tarball commands: can we remove them?

Comment (jpopelka) [reply .packit.yaml]: removed

**Comment by jpopelka** [comment]
> This would be a good time to sync the upstream specfile with the downstream ones.

done, see the second commit

> Would the Fedora PR created by `propose_downstream` override the whole `%changelog` ?

nope, see [PR#2](https://src.fedoraproject.org/rpms/python-dockerfile-parse/pull-request/2#request_diff) as created with `packit propose-update`

> would COPR EPEL builds be useful for us whenever a PR is open?

`rhelbeta-8-x86_64` target added, I don't see any newer `epel` chroot in `copr` yet

**Comment by jpopelka** [comment]
And as @TomasTomecek pointed out today it should be possible to test the Github App even in this PR. Can you [install it](https://packit.dev/packit-as-a-service/) in this repo?

**Comment by MartinBasti** [comment]
^ @rcerven @twaugh only you can install app in this org

**Comment by twaugh** [comment]
Installed.

**Comment by jpopelka** [comment]
So apparently only you can trigger it with `/packit copr-build`.

**Comment by athos-ribeiro** [comment]
Thanks, @jpopelka!!

Can we also build it for `epel-7-x86_64`? From the diff between fedora spec syncs, It seems `python2-six` is being required instead of `python-six` for the rhel cases (and it is also missing in BRs for py2 with tests). I'd say we can also drop the rhel <= 6 conditionals since we no longer support python 2.6.

Finally, the change has one important improvement to the rpm package: we were defining a python2- package but we were generating the %files for a python- package.

**Comment by jpopelka** [comment]
* `epel-7-x86_64` added
* `rhel <= 6` conditional removed

**Comment by jpopelka** [comment]
whoops, sorry, try again please

**Comment by athos-ribeiro** [comment]
Looks great! There only a final detail missing:

> From the diff between fedora spec syncs, It seems python2-six is being required instead of python-six for the rhel cases (and it is also missing in BRs for py2 with tests).

So, for rhel, we should request python-six instead, which is the one available there (also in centos/epel)

**Comment by jpopelka** [comment]
I don't follow, sorry.
Anyway, this PR is about enabling `packit`, not about improving the `spec`.
I've synced the spec with the one in Fedora, if you want any more changes you can either:
- fix it in another PR
- tell me exact diff of what you want to change
- push a commit into my `packit` branch on `jpopelka/dockerfile-parse`

**Comment by jpopelka** [comment]
It's triggered automatically for collaborators (people who can merge a PR).
For the others, you at the moment have to trigger it manually.

**Pull request merged**